packages/cozy-client/src/queries/dsl.js

```diff
@@ -129,6 +129,20 @@ class QueryDefinition {
         Selector: ${JSON.stringify(selector)}`
       )
     }
+
+    const nestedArrayKey = findKey(selector, value => Array.isArray(value))
+    const hasNestedArrayWithoutOperator =
+      nestedArrayKey && !nestedArrayKey.startsWith('$')
+    if (hasNestedArrayWithoutOperator) {
+      throw new Error(
+        `You pass ${JSON.stringify(
+          selector
+        )}, this is a valid mango operation, \n
+           but sift doesn't support it so CozyClient can not evaluate \n
+           the request in memory. You have to use MongoDB operator \n
+           like $in...`
```

Comment:
That's quite a pity, because $in operator forces a full-scan of the index, so this is not something I would recommend :/

Reply:
Damn, I hadn't thought of that. I'm correcting it here, on home and on harvest 🙁

Reply:
Si, ça va être bon dans le cas du worker... Et surtout pas le $or ... 

Parce qu'on doit passer dans cette condition https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/TriggerCollection.js#L130-L143

et donc par une route dédiée et ça semble se baser sur le $in derrière pour que ça marche https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/TriggerCollection.js#L33-L35

Par contre, effectivement avoir mis $and c'est mieux dans l'exemple ;)

Reply:
En fait à la réflexion ça peut pas être un `$and`. Vu que la syntaxe raccourcie fait un $or ou $in je sais plus trop. 

Donc le message d'erreur explicatif doit plutôt être de faire un $or ou $in et que dans ce cas c'est préférable de passer par un partial index.

Reply:
Proposition de formulation: `Use $in or $or operator instead, preferably in a partial index, to avoid costly full-scan`

Reply:
@paultranvan Does the description suit you better?